src/trio/_core/_parking_lot.py

```diff
@@ -234,6 +269,23 @@ def repark_all(self, new_lot: ParkingLot) -> None:
         """
         return self.repark(new_lot, count=len(self))
 
+    def break_lot(self, task: Task) -> None:
+        """Break this lot, causing all parked tasks to raise an error, and any
+        future tasks attempting to park (and unpark? repark?) to error. The error
+        contains a reference to the task sent as a parameter."""
+        self.broken_by = task
+
+        # TODO: is there any reason to use self._pop_several?
+        for parked_task in self._parked:
+            # TODO: weird to phrase this one, we maybe should reraise this error in Lock?
```

Comment:
I agree that it should be reraised -- I think it's best to use a subclass so that anyone using this can catch and reraise without any concern about catching things incorrectly + still have access to the task that caused the parking lot to break.